# HG changeset patch
# User Betterbird <betterbird@betterbird.eu>
# Date 1714504847 -7200
# Parent  81180f2e6e05b8d5ce1ee7d035cd0c63a278c56a
NNN9 - Suppress unneeded logging to the error console.

diff --git a/toolkit/components/extensions/ExtensionChild.sys.mjs b/toolkit/components/extensions/ExtensionChild.sys.mjs
--- a/toolkit/components/extensions/ExtensionChild.sys.mjs
+++ b/toolkit/components/extensions/ExtensionChild.sys.mjs
@@ -669,8 +669,13 @@ export class ProxyAPIImplementation exte
     map.listeners.delete(listener);
     map.ids.delete(id);
     map.removedIds.add(id);
 
+    // The following call throws if there is no actor.
+    // No benefit in calling it.
+    if (!this.childApiManager.conduit.actor) {
+      return;
+    }
     this.childApiManager.conduit.sendRemoveListener({
       childId: this.childApiManager.id,
       listenerId: id,
       path: this.path,
diff --git a/toolkit/components/extensions/ExtensionParent.sys.mjs b/toolkit/components/extensions/ExtensionParent.sys.mjs
--- a/toolkit/components/extensions/ExtensionParent.sys.mjs
+++ b/toolkit/components/extensions/ExtensionParent.sys.mjs
@@ -326,9 +326,11 @@ const ProxyMessenger = {
 
   getTopBrowsingContextId(tabId) {
     // If a tab alredy has content scripts, no need to check private browsing.
     let tab = apiManager.global.tabTracker.getTab(tabId, null);
-    if (!tab || (tab.browser || tab).getAttribute("pending") === "true") {
+    // tab.getAttribute("pending") throws, so the following doesn't work:
+    // if (!tab || (tab.browser || tab).getAttribute("pending") === "true") {
+    if (!tab || !tab.browser || tab.browser.getAttribute("pending") === "true") {
       // No receivers in discarded tabs, so bail early to keep the browser lazy.
       throw new ExtensionError(ERROR_NO_RECEIVERS);
     }
     let browser = tab.linkedBrowser || tab.browser;
@@ -1108,9 +1110,17 @@ ParentAPIManager = {
     }
   },
 
   async recvAPICall(data, { actor }) {
-    let context = this.getContextById(data.childId);
+    let context;
+    // getContextById will throw if the context has been destroyed
+    // in the meantime.
+    try {
+      context = this.getContextById(data.childId);
+    } catch (ex) {
+      console.info(`No context for ${data.childId}`);
+      return;
+    }
     let target = actor.browsingContext?.top.embedderElement;
 
     verifyActorForContext(actor, context);
 
